pkg/internal/vault/vault_test.go

```diff
@@ -146,29 +233,45 @@ func TestSign(t *testing.T) {
 		"a good csr and good response should return a certificate": {
 			csrPEM: csrPEM,
 			issuer: gen.Issuer("vault-issuer",
-				gen.SetIssuerVault(v1.VaultIssuer{}),
+				gen.SetIssuerVault(cmapi.VaultIssuer{}),
 			),
 			fakeClient: vaultfake.NewFakeClient().WithRawRequest(&vault.Response{
 				Response: &http.Response{
 					Body: ioutil.NopCloser(bytes.NewReader(bundleData))},
 			}, nil),
 			expectedErr:  nil,
 			expectedCert: testCertBundle,
-			expectedCA:   testCertBundle,
+			expectedCA:   testIntermediateCa,
 		},
 
 		"vault issuer with namespace specified": {
 			csrPEM: csrPEM,
 			issuer: gen.Issuer("vault-issuer",
-				gen.SetIssuerVault(v1.VaultIssuer{Namespace: "test"}),
+				gen.SetIssuerVault(cmapi.VaultIssuer{Namespace: "test"}),
 			),
 			fakeClient: vaultfake.NewFakeClient().WithRawRequest(&vault.Response{
 				Response: &http.Response{
 					Body: ioutil.NopCloser(bytes.NewReader(bundleData))},
 			}, nil),
 			expectedErr:  nil,
 			expectedCert: testCertBundle,
-			expectedCA:   testCertBundle,
+			expectedCA:   testIntermediateCa,
```

Comment:
certificate chain should be the expected CA here

Reply:
I don't think that's correct - the `ca.crt` should contain the *root certificate* as the entity utilising the leaf should be presenting the chain up to the trusted root to whatever it is connecting to/having presented to it